File: tasks/remove_demo_classes.mjs
import { readdirSync, readFileSync, writeFileSync } from 'fs';
import { join } from 'path';

const VIEWS_DIR = String.raw`C:\Users\sxp267\IdeaProjects\holon-vaadin-flow\demo\src\main\java\com\holonplatform\vaadin\flow\demo\ui\views`;

// Matches a whole line whose only content is an addClassName("demo-...") call.
// Handles optional receiver (e.g. "title.", "this."), optional trailing CR (\r\n files).
const LINE_RE = /^[ \t]*(?:\w[\w.]*\.)?addClassName\("demo-[^"]*"\);[ \t]*\r?$/;

let totalRemoved = 0, filesChanged = 0;

for (const name of readdirSync(VIEWS_DIR).filter(f => f.endsWith('.java'))) {
  const file = join(VIEWS_DIR, name);
  const lines = readFileSync(file, 'utf8').split('\n');
  const kept = lines.filter(l => !LINE_RE.test(l));
  const removed = lines.length - kept.length;
  if (removed > 0) {
    writeFileSync(file, kept.join('\n'), 'utf8');
    console.log(`  ${name}: removed ${removed} line(s)`);
    totalRemoved += removed;
    filesChanged++;
  }
}

console.log(`\nDone. ${totalRemoved} line(s) removed across ${filesChanged} file(s).`);

File: tasks/remove_demo_classes.py
"""
Removes all lines that contain addClassName("demo-  from every Java file
under demo/src/main/java/.../views/.

Also removes addClassName calls that set a "demo-" class on 'this'
via the bare form: addClassName("demo-...").

A line is removed only if its sole purpose is the addClassName call
(i.e. the line, when stripped, matches the pattern).
Lines that combine addClassName with another expression on the same line
are left untouched (safety guard).
"""
import re
import pathlib

VIEWS_DIR = pathlib.Path(
    r"C:\Users\sxp267\IdeaProjects\holon-vaadin-flow\demo\src\main\java"
    r"\com\holonplatform\vaadin\flow\demo\ui\views"
)

# Pattern: optional leading whitespace, optional receiver + dot, addClassName("demo-..."), optional semicolon, optional trailing whitespace
DEMO_CLASS_LINE = re.compile(r'^\s*\S*\.?addClassName\s*\(\s*"demo-[^"]*"\s*\)\s*;?\s*$')

removed_total = 0
files_changed = 0

for java_file in sorted(VIEWS_DIR.glob("*.java")):
    original = java_file.read_text(encoding="utf-8")
    lines = original.splitlines(keepends=True)
    kept = []
    removed_here = 0
    for line in lines:
        if DEMO_CLASS_LINE.match(line):
            removed_here += 1
        else:
            kept.append(line)
    if removed_here:
        java_file.write_text("".join(kept), encoding="utf-8")
        print(f"  {java_file.name}: removed {removed_here} line(s)")
        removed_total += removed_here
        files_changed += 1

print(f"\nDone. {removed_total} line(s) removed across {files_changed} file(s).")

